### DISCUSSION TITLE
Trying to use 3.5 turbo

### DISCUSSION THREAD

**Discussion by rdecosta** [srcnalt/OpenAI-Unity#41] [General]
I'm trying to use 3.5-turbo, but there's no obvious method for setting this. If I use the url OpenAI suggests (https://api.openai.com/v1/chat/gpt-3.5-turbo/completions) I get Error Message: Cannot specify both model and engine

**Comment by srcnalt** [comment]
Hi @rdecosta, the endpoint you describe does not exist in that form. The chat completion endpoint is `https://api.openai.com/v1/chat/completions` And the names of models that could be used are listed here. https://platform.openai.com/docs/models/model-endpoint-compatibility

Using this plugin you can make a request such as:

```c#
var req = new CreateChatCompletionRequest
{
  Model = "gpt-3.5-turbo",
  Messages = new List<ChatMessage>()
  {
    new ChatMessage() { Role = "user", Content = "Hello!" }
  }
};
var res = await openai.CreateChatCompletion(req);
```

**Reply by rdecosta** [reply]
If I change the hard-coded (?) endpoint to the one you suggest, I get Error Message: Invalid URL (POST /v1/chat/completions/completions)

the code snippet above doesn't seem to make allowance for changing the endpoint. Unless I'm missing something really obvious.

**Reply by srcnalt** [reply]
The plugin covers all the endpoints, you do not need to change anything in this case.
You can check OpenAiApi.cs and all the possible calls are covered there.

https://github.com/srcnalt/OpenAI-Unity/blob/31aaaa4b471b9f0300256f933ac94a10eebb4ef7/Runtime/OpenAIApi.cs#L251
